src/shared/vscode/commands2.ts

```diff
@@ -18,6 +18,34 @@ import { keysAsInt } from '../utilities/tsUtils'
 type Callback = (...args: any[]) => any
 type CommandFactory<T extends Callback, U extends any[]> = (...parameters: U) => T
 
+/**
+ * HACK
+ *
+ * If a command matches the following conditions:
+ * - Is registered with VS Code
+ * - Has args for the execution of the command
+ * - Can be executed through a package.json contribution (eg: ellipsis menu in View)
+ *
+ * Then unexpected args will be passed to the command.
```

Comment:
It's not really unexpected. Any command called from a treeview can get a treenode, etc. The commands simply need to be defined with parameters that express that :)